homeassistant/components/discovergy/sensor.py

```diff
@@ -209,13 +207,15 @@ def __init__(
         self.data_key = data_key
 
         self.entity_description = description
-        self._attr_unique_id = f"{meter.full_serial_number}-{data_key}"
+        self._attr_unique_id = f"{coordinator.meter.full_serial_number}-{data_key}"
         self._attr_device_info = DeviceInfo(
-            identifiers={(DOMAIN, meter.meter_id)},
-            name=f"{meter.measurement_type.capitalize()} {meter.location.street} {meter.location.street_number}",
-            model=meter.type,
+            identifiers={(DOMAIN, coordinator.meter.meter_id)},
+            name=f"{coordinator.meter.measurement_type.capitalize()} "
+            f"{coordinator.meter.location.street} "
+            f"{coordinator.meter.location.street_number}",
+            model=coordinator.meter.type,
             manufacturer=MANUFACTURER,
-            serial_number=meter.full_serial_number,
+            serial_number=coordinator.meter.full_serial_number,
         )
```

Comment:
Since you're accessing `coordinator.meter` quite a lot, maybe just add a variable where you access it, so you only access it once

Reply:
Good point, thanks 👍